[NO QA] Pass personal policyID to InboxCallUser

jasperhuangg opened the pull request:
### Details
We're adding guides calls to Expensify.cash. Since we currently don't have the concept of an active policy/workspace, we'll default to passing the user's personal policyID to `Inbox_CallUser`. Otherwise, the job will throw an exception once it starts running in Expensiworks.

### Fixed Issues
Related to https://github.com/Expensify/Expensify/issues/167905

### Tests
N/A (will be tested with https://github.com/Expensify/Web-Expensify/pull/31319)

### QA Steps
N/A

jasperhuangg committed:
Pass personal policyID to InboxCallUser

jasperhuangg committed:
Update comment

jasperhuangg committed:
fix parameter access

Jag96 commented on src/pages/RequestCallPage.js:
```diff
@@ -70,7 +79,8 @@ class RequestCallPage extends Component {
         }
 
-        requestConciergeDMCall('', this.state.firstName, this.state.lastName, this.state.phoneNumber)
+        const personalPolicyID = _.find(this.props.policies, policy => policy.type === 'personal').id;
```
Also, what happens if `find` doesn't return a value here? Not sure how offline-friendly this needs to be, but do we need to check that value before trying to access ID and perhaps add a generic error message if we can't find a personal policy?

jasperhuangg replied:
Good thinking! I'll add some type of warning if we can't find a personal policy.

jasperhuangg committed:
Check if personal policy can be found before submitting form

jasperhuangg commented:
@Jag96 Thanks for the review! Addressed some of your concerns, should be good for another review. 

This'll be on hold for now since I'm waiting on a copy-check for the new text that'll go into the you mentioned.

Jag96 commented on src/pages/RequestCallPage.js:
```diff
@@ -65,22 +74,27 @@ class RequestCallPage extends Component {
         this.setState({isLoading: true});
         if (!this.state.firstName.length || !this.state.lastName.length) {
-            Growl.show(this.props.translate('requestCallPage.growlMessageEmptyName'), CONST.GROWL.ERROR);
+            Growl.success(this.props.translate('requestCallPage.growlMessageEmptyName'));
```
Doesn't look like `success` is an existing function for growl yet, same for other usage below

jasperhuangg replied:
Ah duh 😓 just assumed it existed since error was there too. I'll go ahead and implement it

jasperhuangg committed:
implement Growl.success

jasperhuangg committed:
Merge branch 'main' of github.com:Expensify/Expensify.cash into jasper-requestCallPersonalPolicy

Jag96 reviewed (commented):
LGTM! Will approve once `growlMessageNoPersonalPolicy` has a copy set

jasperhuangg committed:
Merge branch 'main' of github.com:Expensify/Expensify.cash into jasper-requestCallPersonalPolicy

jasperhuangg committed:
update translation

jasperhuangg commented:
@Jag96 I've updated the translation and taken it off hold, feel free to merge if you think everything checks out, thanks!

Jag96 reviewed (approved):
LGTM!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.
